src/methods/geom_relations/geom_geom_processors.jl

```diff
@@ -0,0 +1,739 @@
+    closed_curve = false,
+)
+    # Set up requirments
+    in_req_met = !in_require
+    on_req_met = !on_require
+    out_req_met = !out_require
+    # Determine curve endpoints
+    nl = GI.npoint(line)
+    nc = GI.npoint(curve)
+    first_last_equal_line = equals(GI.getpoint(line, 1), GI.getpoint(line, nl))
+    first_last_equal_curve = equals(GI.getpoint(curve, 1), GI.getpoint(curve, nc))
+    nl -= first_last_equal_line ? 1 : 0
+    nc -= first_last_equal_curve ? 1 : 0
+    closed_line |= first_last_equal_line
+    closed_curve |= first_last_equal_curve
+    # Loop over each line segment
+    l_start = GI.getpoint(line, closed_line ? nl : 1)
+    i = closed_line ? 1 : 2
+    while i ≤ nl
+        l_end = GI.getpoint(line, i)
+        c_start = GI.getpoint(curve, closed_curve ? nc : 1)
+        # Loop over each curve segment
+        for j in (closed_curve ? 1 : 2):nc
+            c_end = GI.getpoint(curve, j)
+            # Check if line and curve segments meet
+            seg_val = _segment_segment_orientation(
+                (l_start, l_end),
+                (c_start, c_end),
+            )
+            # If segments are co-linear
+            if seg_val == line_over
+                !over_allow && return false
+                # at least one point in, meets requirments
+                in_req_met = true
+                point_val = _point_segment_orientation(
+                    l_start,
+                    c_start, c_end,
+                )
+                # If entire segment isn't covered, consider remaining section
+                if point_val != point_out
+                    if _point_segment_orientation(
+                        l_end,
+                        c_start, c_end,
+                    ) != point_out
+                        l_start = l_end
+                        i += 1
+                        break
+                    elseif _point_segment_orientation(
+                        c_start,
+                        l_start, l_end,
+                    ) != point_out && !equals(l_start, c_start)
+                        l_start = c_start
+                        break
+                    elseif _point_segment_orientation(
+                        c_end,
+                        l_start, l_end,
+                    ) != point_out && !equals(l_start, c_end)
+                        l_start = c_end
+                        break
+                    end
+                end
+            else
+                if seg_val == line_cross
+                    !cross_allow && return false
+                    in_req_met = true
+                elseif seg_val == line_hinge  # could cross or overlap
+                    # Determine location of intersection point on each segment
+                    _, fracs = _intersection_point(
+                        (_tuple_point(l_start), _tuple_point(l_end)),
+                        (_tuple_point(c_start), _tuple_point(c_end))
+                    )
+                    (α, β) =
```

Comment:
You could tab the rest of the if/else block back if you do this:
```suggestion
                    (α, β) = if !isnothing(fracs)
```